mlflow/langchain/langchain_tracer.py

```diff
@@ -111,13 +122,14 @@ def _end_span(
         status=SpanStatus(SpanStatusCode.OK),
     ):
         """Close MLflow Span (or Trace if it is root component)"""
-        self._mlflow_client.end_span(
-            request_id=span.request_id,
-            span_id=span.span_id,
-            outputs=outputs,
-            attributes=attributes,
-            status=status,
-        )
+        with set_prediction_context(self._prediction_context):
```

Comment:
note: Both `start_trace` and `end_span` call needs to be executed within the context, since they depend on the context var.